tests/functional/olp-cpp-sdk-dataservice-read/DataserviceReadVolatileLayerClientTest.cpp

```diff
@@ -224,4 +341,273 @@ TEST_F(VolatileLayerClientTest, GetPartitionsDifferentFetchOptions) {
   }
 }
 
+TEST_F(VolatileLayerClientTest, Prefetch) {
+  const auto kTileId = GetTestTile();
+  auto layer = CustomParameters::getArgument(kPrefetchLayer);
+  auto catalog = CustomParameters::getArgument(kPrefetchCatalog);
+
+  PrepareForPrefetch();
+
+  olp::client::HRN hrn(catalog);
+  VolatileLayerClient client(hrn, layer, settings_);
+  {
+    SCOPED_TRACE("Prefetch tiles online and store them in memory cache");
+    std::vector<olp::geo::TileKey> tile_keys = {
+        olp::geo::TileKey::FromHereTile(kTileId)};
+
+    auto request = olp::dataservice::read::PrefetchTilesRequest()
+                       .WithTileKeys(tile_keys)
+                       .WithMinLevel(10)
+                       .WithMaxLevel(12);
+
+    std::promise<PrefetchTilesResponse> promise;
+    std::future<PrefetchTilesResponse> future = promise.get_future();
+    auto token = client.PrefetchTiles(
+        request, [&promise](PrefetchTilesResponse response) {
+          promise.set_value(std::move(response));
+        });
+
+    ASSERT_NE(future.wait_for(kTimeout), std::future_status::timeout);
+    PrefetchTilesResponse response = future.get();
+    EXPECT_TRUE(response.IsSuccessful());
+    ASSERT_FALSE(response.GetResult().empty());
+
+    const auto& result = response.GetResult();
+
+    for (auto tile_result : result) {
+      EXPECT_TRUE(tile_result->IsSuccessful());
+      ASSERT_TRUE(tile_result->tile_key_.IsValid());
```

Comment:
But you should know which tiles you expect. Why do you only test if it is valid when you should be testing that expected tiles have been downloaded. Please check the tiles that were downloaded.